fanficfare/adapters/adapter_asianfanficscom.py

```diff
@@ -138,18 +155,26 @@ def doExtractChapterUrlsAndMetadata(self,get_cover=True):
         soup = self.make_soup(data)
 
         # it is best to log in whenever possible, unless already logged in from cache..
-        if self.password or self.getConfig("password") and "Logout" not in data:
+        loginCheck = soup.find('div',{'id':'login'})
+        if self.password or self.getConfig("password") and loginCheck:
             self.performLogin(url,soup)
             data = self._fetchUrl(url,usecache=False)
             soup = self.make_soup(data)
-        elif "Logout" not in data:
-            logger.info('Note: Logging in is highly recommended, as this website censors text if not logged in.')
+        elif loginCheck:
+            logger.info('Note: Logging in is highly recommended, as this website censors text and removes certain HTML tags if not logged in.')
 
         # adult check
         self.checkSoup = self.doAdultCheck(url,soup)
         if self.checkSoup:
             soup = self.checkSoup
 
+        # subscription check
+        loginCheck = soup.find('div',{'id':'login'})
+        if self.getConfig("auto_sub") and not loginCheck:
+            self.subSoup = self.doSubCheck(url,soup)
```

Comment:
Again, if `doSubCheck` (after suitable renaming) always returns `soup` either changed or unchanged, there's no need to check it.
Also, why `self.subSoup`?  I don't see any reason for that to be saved in `self` if it's not used again?